code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -215,6 +231,40 @@ func (msc *MinerSmartContract) GetSharderListHandler(ctx context.Context, params
 	}, nil
 }
 
+// GetSharderGeolocationsHandler returns a list of all sharders latitude and longitude
+func (msc *MinerSmartContract) GetSharderGeolocationsHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
+	var (
+		offsetString = params.Get("offset")
+		limitString  = params.Get("limit")
+		activeString = params.Get("active")
+	)
+
+	offset, limit, err := getOffsetLimitParam(offsetString, limitString)
+	if err != nil {
+		return nil, err
+	}
+
+	filter := event.SharderQuery{}
+	if activeString != "" {
+		active, err := strconv.ParseBool(activeString)
+		if err != nil {
+			return nil, common.NewErrBadRequest("active parameter is not valid")
+		}
+		filter.Active = null.BoolFrom(active)
+	}
+
+	if balances.GetEventDB() == nil {
+		return nil, common.NewErrInternal("can't get sharders. no db connection")
```

Comment:
The same as above, check if event db is nil before parsing the params

Reply:
done